okio-fakefilesystem/src/commonMain/kotlin/okio/fakefilesystem/FakeFileSystem.kt

```diff
@@ -218,8 +238,13 @@ class FakeFileSystem(
     if (element !is File) {
       throw IOException("not a file: $file")
     }
+    if (!allowReadsWhileWriting) {
+      openFileOrNull(canonicalPath, operation = WRITE)?.let {
+        throw IOException("file is already open for writing $file", it.backtrace)
+      }
+    }
 
-    val openFile = OpenFile(canonicalPath, Exception("file opened for reading here"))
+    val openFile = OpenFile(canonicalPath, READ, Exception("file opened for reading here"))
     openFiles += openFile
```

Comment:
Should this entire method be synchronized? I don't think this implementation is designed to be highly concurrent, and 100% predictability probably beats any chance at race conditions in tests?

Reply:
Yeah, I gotta do that. For JVM it’s easy, for native? Unclear.